### PR TITLE
Add migration for the addons.contributions field

### LINKED ISSUE #9842: Add migration for the `addons.contributions` field to be in sync with the django model
The `Addon` django model defines the `contributions` field with `blank=True` but not with `null=True`, which means this field is not nullable. Yet, the -dev, -stage and likely -prod DBs have the following definition:

```
MySQL [addons_dev_allizom_org]> DESCRIBE addons;
+---------------------+---------------------+------+-----+---------------------+----------------+
| Field               | Type                | Null | Key | Default             | Extra          |
+---------------------+---------------------+------+-----+---------------------+----------------+
| contributions       | varchar(255)        | YES  |     | NULL                |                |
+---------------------+---------------------+------+-----+---------------------+----------------+
```

In other words, the `contributions` field is nullable on the hosted envs.

We should change this definition so that the field is not nullable anymore (i.e. the definition should match the django model). We can do it by:

- setting all `NULL` values to `''` (empty string)
- modifying the field definition

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #9842

### CODE REVIEW

File: src/olympia/migrations/1052-remove-nullable-constraint-on-addon-contributions.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Change NULL values to empty strings.
+UPDATE `addons` SET `contributions` = '' WHERE `contributions` IS NULL;
+
+-- Add NOT NULL constraint.
+ALTER TABLE `addons` MODIFY `contributions` VARCHAR(255) DEFAULT '' NOT NULL;
```
Comment (willdurand) [inline src/olympia/migrations/1052-remove-nullable-constraint-on-addon-contributions.sql]: I set `''` as a default value but it seems django would no set anything (default is `NULL`). Should I keep it or remove it?

Comment (diox) [reply src/olympia/migrations/1052-remove-nullable-constraint-on-addon-contributions.sql]: To keep it in line with Django, do not add `DEFAULT ''`. Django will store an empty string itself everytime you use the model, per its definition.

Comment (willdurand) [reply src/olympia/migrations/1052-remove-nullable-constraint-on-addon-contributions.sql]: Ok, I was not sure about that and could not find the info in the Django doc.

**Comment by willdurand** [comment]
A job is failing because of a locale issue.

**Comment by diox** [comment]
Do add to the push notes that we have a migration affecting the addons table, ops do special steps when that happens to prevent versioncheck breakage.

**Comment by willdurand** [comment]
> Do add to the push notes that we have a migration affecting the addons table, ops do special steps when that happens to prevent versioncheck breakage.

ok, thanks!

**Pull request merged**